apps/api/src/controllers/competition.controller.ts

```diff
@@ -479,69 +483,66 @@ export function makeCompetitionController(services: ServiceRegistry) {
             status,
+            // Default limit 10, max 100. It's important we don't call this without a limit
             pagingParams,
           );
 
         // If user is authenticated, enrich competitions with voting information
         let enrichedCompetitions = competitions;
         if (userId) {
-          enrichedCompetitions = await Promise.all(
-            competitions.map(async (competition) => {
-              try {
-                // Get voting state for this user and competition
-                const votingState =
-                  await services.voteManager.getCompetitionVotingState(
-                    userId,
-                    competition.id,
-                  );
-
-                // Get total vote counts for this competition (for display purposes)
-                const voteCountsMap =
-                  await services.voteManager.getVoteCountsByCompetition(
-                    competition.id,
-                  );
-                const totalVotes = Array.from(voteCountsMap.values()).reduce(
-                  (sum, count) => sum + count,
-                  0,
-                );
-
-                // Get trading constraints and rewards for this competition
-                const tradingConstraintsRaw =
-                  await services.tradingConstraintsService.getConstraints(
-                    competition.id,
-                  );
-                const tradingConstraints = {
-                  minimumPairAgeHours:
-                    tradingConstraintsRaw?.minimumPairAgeHours,
-                  minimum24hVolumeUsd:
-                    tradingConstraintsRaw?.minimum24hVolumeUsd,
-                  minimumLiquidityUsd:
-                    tradingConstraintsRaw?.minimumLiquidityUsd,
-                  minimumFdvUsd: tradingConstraintsRaw?.minimumFdvUsd,
-                };
-
-                return {
-                  ...competition,
-                  tradingConstraints,
-                  votingEnabled:
-                    votingState.canVote || votingState.info.hasVoted,
-                  userVotingInfo: votingState,
-                  totalVotes,
-                };
-              } catch (error) {
-                // If there's an error getting vote data, just return the competition without vote info
-                competitionLogger.warn(
-                  `Failed to get vote data for competition ${competition.id}:`,
-                  error,
-                );
-                return {
-                  ...competition,
-                  votingEnabled: false,
-                  totalVotes: 0,
-                };
-              }
-            }),
+          const competitionIds = competitions.map((c) => c.id);
+
+          // Fetch all data in parallel with batch queries (2 queries total instead of 4N)
+          const [enrichmentData, voteCountsMap] = await Promise.all([
+            getEnrichedCompetitions(userId, competitionIds),
```

Comment:
`Promise.all` is just two methods, and they both only do one query no matter what.